### DISCUSSION TITLE
Is `getContainerOfType()` Still Exported?

### DISCUSSION THREAD

**Discussion by ballcoach12** [eclipse-langium/langium#1448] [Q&A] [answered]
I am taking a dry run at upgrading from 2.0.0 to 3.0.0 in my project, and it seems that `getContainerOfType()` and `streamAllContents()` are not available via an import, even though they appear to be specified on the module export for Langium.  

Am I missing something here or will I need to reference it directly through the path to the `node_modules` folder?

**Comment by msujew** [comment, marked as answer]
Hey @ballcoach12,

We are now exporting these functions via dedicated namespaces, most important ones are available through the `AstUtils` namespace. See also [the breaking changes of 3.0](https://github.com/eclipse-langium/langium/blob/main/packages/langium/CHANGELOG.md#breaking-changes).

**Reply by ballcoach12** [reply]
Got it.  Thanks!
